driver/pgdriver/config.go

```diff
@@ -246,15 +248,66 @@ func parseDSN(dsn string) ([]Option, error) {
 		opts = append(opts, WithApplicationName(appName))
 	}
 
-	switch sslMode := q.string("sslmode"); sslMode {
-	case "verify-ca", "verify-full":
-		opts = append(opts, WithTLSConfig(new(tls.Config)))
-	case "allow", "prefer", "require", "":
-		opts = append(opts, WithTLSConfig(&tls.Config{InsecureSkipVerify: true}))
-	case "disable":
-		opts = append(opts, WithInsecure(true))
-	default:
-		return nil, fmt.Errorf("pgdriver: sslmode '%s' is not supported", sslMode)
+	if sslMode, sslRootCert := q.string("sslmode"), q.string("sslrootcert"); sslMode != "" || sslRootCert != "" {
+		tlsConfig := &tls.Config{}
+		switch sslMode {
+		case "disable":
+			tlsConfig = nil
+		case "allow", "prefer", "":
```

Comment:
`allow` and `prefer` were not, and are still not entirely correct.

Specifically, if `allow` is used, the initial connection attempt should be done without TLS, and only if that fails, a TLS connection should be attempted.
If `prefer` is used, the reverse should happen (first with TLS, then without).

As the current semantics of `Config` doesn't really allow for these fallback semantics, I've avoided refactoring more stuff in this PR.

Reply:
Thanks for the information - I was not aware of that... I guess we can discuss this in detail later if there is interest in making this work.